prophet/handlers.go

```diff
@@ -19,7 +20,22 @@ type FutureHandler interface {
 }
 
 // Execute executes a given future, by invoking the registered handler.
-func Execute(ctx context.Context, f Future) (FutureResult, error) {
+func Execute(ctx context.Context, f Future) (res FutureResult, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			switch x := r.(type) {
+			case string:
+				err = fmt.Errorf("future execution failed with panic: %s", x)
+			case error:
+				err = fmt.Errorf("future execution failed with panic: %w", x)
+			default:
+				err = errors.New("future execution failed with unknown panic")
+			}
+
+			res = FutureResult{}
```

Comment:
I'd just do this:

```suggestion
			err = fmt.Errorf("panic executing future: %v", r)
			res = FutureResult{}
```

wrapping the error coming from a panic is probably not very helpful, since it means that someone forgot to properly handle an error by returning it